core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1647,11 +1647,13 @@ class Partition(val topicPartition: TopicPartition,
     * @param newOffset The new offset to start the log with
     * @param isFuture True iff the truncation should be performed on the future log of this partition
     */
-  def truncateFullyAndStartAt(newOffset: Long, isFuture: Boolean): Unit = {
+  def truncateFullyAndStartAt(newOffset: Long,
+                              isFuture: Boolean,
+                              logStartOffsetOpt: Option[Long] = None): Unit = {
```

Comment:
Please add javadoc for this new parameter. Same comment at other places like `LogManager` and `UnifiedLog`.